graphql_compiler/cost_estimation/analysis.py

```diff
@@ -130,10 +130,32 @@ def get_distinct_result_set_estimates(
             schema_info, filter_infos, parameters, vertex_type_name, class_count
         )
 
-    # TODO(bojanserafimov): If there's a many-to-one edge from A to B in the query, the
-    #                       distinct result set estimate for B cannot be greater than
-    #                       the one for A. Taking this into account would make the results
-    #                       more accurate.
+    single_destination_traversals = set()
+    for location, _ in query_metadata.registered_locations:
+        if not isinstance(location, Location):
+            # TODO(bojanserafimov): We currently ignore FoldScopeLocations. However, a unique
+            #                       filter on a FoldScopeLocation also uniquely filters the
+            #                       enclosing scope.
+            continue
+
+        if len(location.query_path) > 1:
+            from_path = location.query_path[:-1]
+            to_path = location.query_path
+            edge_direction, edge_name = get_edge_direction_and_name(location.query_path[-1])
+            edge_constraints = schema_info.edge_constraints.get(edge_name, EdgeConstraint(0))
```

Comment:
I am not sure what this syntax `EdgeConstraint(0)` does, but this seems sketchy. Do you mind explaining or fixing?

Reply:
Flags are represented with bits. 0 means all bits unset. Can't find another way to get this value.